spark/src/main/scala/geotrellis/spark/io/hadoop/geotiff/InMemoryGeoTiffAttributeStore.scala

```diff
@@ -3,7 +3,7 @@ package geotrellis.spark.io.hadoop.geotiff
 import geotrellis.vector.ProjectedExtent
 import java.net.URI
 
-case class InMemoryGeoTiffAttributeStore(getData: () => GeoTiffMetadataTree[GeoTiffMetadata]) extends CollectionAttributeStore[GeoTiffMetadata] {
+abstract class InMemoryGeoTiffAttributeStore(getData: () => GeoTiffMetadataTree[GeoTiffMetadata]) extends CollectionAttributeStore[GeoTiffMetadata] {
```

Comment:
Making this class `abstract` makes sense, why not move `getData` into abstract method in that case, just like `persist` ?

Reply:
Its kind of weird that `GeoTiffMetadataTree` is exposed as part of the interface here. Does it need to be. Perhaps it can just be used privately to build an index from incoming list of `GeoTiffMetadata`

Reply:
@echeipesh it's more an oversight (`GeoTiffMetadataTree`)